oat-sa/extension-pcisample
# Fix/TR-1240/Restore the previous version of text reader aside the new one

Related to: https://oat-sa.atlassian.net/browse/TR-1240

Restore the previous version of the textReader interaction aside the new version.
The purpose is to allow existing items to continue working without having to fix them all one by one.

> **Note:** the code for the PCI has been restored from its previous state before it was deleted. So no need to review it in details. This is a legacy code, but we don't need to update/fix it, it was done already by the refactoring to the IMS standard.

How to test (sorry for the monkey job):

- **Update**
    1. install a TAO instance with the pciSamples extension from `v3.0.1`.
    2. import the test `test_textReader_v0.9.0.zip` from the ticket
    3. publish the test, and check it can be passed well
    4. check the preview of the item is ok
    5. checkout the tag `v3.1.0`
    6. update TAO
    7. check the test cannot be completed
    8. check the test cannot be exported
    9. checkout the branch of this PR
    10. update TAO: the update should go well.
    11. import the test `test_textReader_v1.0.0.zip` from the ticket
    12. publish the test, and check it can be passed well
    13. check the test of the PCI in v0.9.0 can be passed well
    14. check the URL to get the PCI is different for both (in the network panel, check the URL to get the textReader runtime)
    15. check the tests can be exported

- **Install**
    1. install a TAO instance with the branch checked out on pciSamples: the install should go well
    2. import the test `test_textReader_v0.9.0.zip` from the ticket
    3. import the test `test_textReader_v1.0.0.zip` from the ticket
    4. publish the tests, and check they can be passed well
    5. check the URL to get the PCI is different for both (in the network panel, check the URL to get the textReader runtime)
    6. check the preview of the items are ok
    7. check the tests can be exported

**Note:** to check the code is present in both version, you can check the content of the `data` folder:
- manifest for the IMS version: `data/qtiItemImsPci`
- manifest for the OAT version: `data/qtiItemPci`
- registered code (each will have 2 folders): `data/portableElementStorage`
